fence/blueprints/data/indexd.py

```
@@ -495,6 +495,7 @@ def delete_files(self, urls=None, delete_all=True):
                 )
             )
             return location.delete(bucket, file_suffix)
```

hey, this function doesn't look right, it `returns` after deleting the first `location` 😱

---

@mattreex what do you think of this?
```
response = ("No URLs to delete", 200)
for location in locations_to_delete:
    response = location.delete(...)
    if something went wrong, break
return response
```

---

Yeah I thought that was strange as well, but I thought someone would surely have pointed it out by now if this only ever deleted one url. But since you pointed it out too I'll look into it.

---

Guess it wasn't that long ago :) https://github.com/uc-cdis/fence/pull/776 
I think the gist will work, but then only the last response is sent. I think we could make it a generator function...

---

i'd say it's fine to only return the last response if they are all 2xx... As long as we don't ignore error responses

---

and you can update the function's docstring `return` line